auditlog/models.py

```diff
@@ -208,6 +215,61 @@ def _get_pk_value(self, instance):
             pk = self._get_pk_value(pk)
         return pk
 
+    def _get_serialized_data_or_none(self, instance):
+        from .registry import auditlog
+
+        opts = auditlog.get_serialize_options(instance.__class__)
+        if not opts["serialize_data"]:
+            return None
+
+        model_fields = auditlog.get_model_fields(instance.__class__)
+        kwargs = opts.get("serialize_kwargs", {})
+
+        if opts["serialize_auditlog_fields_only"]:
+            kwargs.setdefault(
+                "fields", self._get_applicable_model_fields(instance, model_fields)
+            )
+
+        data = dict(json.loads(serializers.serialize("json", (instance,), **kwargs))[0])
+
+        if model_fields["mask_fields"]:
+            data = self._mask_serialized_fields(data, model_fields)
+
+        return data
+
+    def _get_applicable_model_fields(
+        self, instance, model_fields: Dict[str, List[str]]
+    ) -> List[str]:
+        include_fields = model_fields["include_fields"]
+        exclude_fields = model_fields["exclude_fields"]
+        all_field_names = [field.name for field in instance._meta.fields]
+
+        if not include_fields and not exclude_fields:
+            return all_field_names
+
+        include_fields = include_fields or all_field_names
+        filtered_field_names = set(all_field_names).intersection(include_fields)
+        filtered_field_names = filtered_field_names.difference(exclude_fields)
```

Comment:
I think we don't need an intersection between `all_field_names` and `include_fields` because it's already happening in line before. am I wrong?
```suggestion
        filtered_field_names = set(filtered_field_names).difference(exclude_fields)
```

Reply:
Good eye. You are again correct. We can remove a few lines of logic there.